landingai/pipeline/image_source.py

```diff
@@ -368,8 +368,14 @@ def _detect_motion(self, frame: np.ndarray) -> bool:  # TODO Needs test cases
     def __iter__(self) -> Any:
         return self
 
-    def __next__(self) -> "FrameSet":
-        return self.get_latest_frame()
+    def __next__(self) -> "Frame":
+        latest_frame = self.get_latest_frame()
+        while latest_frame is None:
+            # If motion detection is set, we may get empty frames.
+            # We should not yield then, though: we should wait for the next frame
+            # until we get a real Frame.
+            latest_frame = self.get_latest_frame()
+        return latest_frame
```

Comment:
Nice!  I'm wondering whether it should be optional or not.... on one hand `get_latest_frame` is already blocking BUT the caller knows that it will return in 1/FPS..... Now with this change it may "never" return

Reply:
Sometimes the pattern (inspired from OpenCV) is to "busy wait"  and check also for key presses to about the acquisition loop (FYI: I scanned our code and we are not doing it).... How can we enable customers to break this acquisition loop?

Agree about extracting motion detection and sampling out..... motion detection requires state which we don't keep anywhere else (e.g. Frames) this is why it got bundled with the NetwrokedCamera.....   about sampling ...it has its tricks too..in the case of networked camera we need a thread to fetch frames but then we control what to decode (to save CPU)....   In brief not easy but definitely worth it if we can find a way to do it  (BTW: we have frame sampling logic when reading video files too)

Reply:
They way to break from the acquisition loop is the same as any infinite loop: a `ctrl + c` to raise `KeyboardInterrupt`,  or add a counter and a `break` statement, etc.  This would be the same pattern either if we yield something or we block the iterator.

About waiting on key press and things like that, it is useful for quick experiments and Collab/Notebook data science tests, but we need to keep in mind software engineers integrating the SDK directly in their backend too. In those cases, infinite loop yielding frames with no extra human intervention is the expected behavior. The less burden and cognitive load we put on them to check invalid frames, the better.

I see that refactoring this would be a bigger effort. We should leave it for another future PR.